src/ripple/rpc/handlers/LedgerData.cpp

```diff
@@ -48,15 +48,16 @@ Json::Value doLedgerData (RPC::Context& context)
     if (!lpLedger)
         return jvResult;
 
+    bool const isMarker = params.isMember (jss::marker);
     boost::optional<ReadView::key_type> key = ReadView::key_type();
-    if (params.isMember (jss::marker))
+    if (isMarker)
```

Comment:
This isn't related to this PR, but since I noticed: can someone explain to me why `key` is a `boost::optional`? It doesn't seem necessary.